contents/swift/welcome-to-swift.md

```diff
@@ -126,24 +174,67 @@ Next, we introduce the structs that conform to the protocols above.
 ```swift
 // Penguins can't fly ):
 struct Penguin: Bird {
-    let name: String
-    let canFly = false
+    var canFly: Bool {
+        return false
+    }
 }
 
 struct Eagle: Bird, Flyable {
-    let name: String
-    let canFly = true
+    var canFly: Bool {
+        return true
+    }
  
     var airspeedVelocity: Double {
         return 160.0
-  }
+    }
 }
 ```
 
 And if you haven't noticed, protocols are extremely similar to interfaces in Java. 
 
 To understand more about POP, watching this [WWDC 2015 talk](https://www.youtube.com/watch?v=g2LwFZatfTI) is highly recommended.
 
+## Extensions
+
+Extensions add new functionality to an existing class, structure, enumeration, or protocol type. Using the example of `Eagle` above, we can segment the code neatly by using extensions to implement each protocol separately:
+
+```swift
+struct Eagle {
+    // you can add class specific behavior here
+}
+
+extension Eagle: Bird {
+    var canFly: Bool {
+        return true
+    }
+}
+    
+extension Eagle: Flyable {
+    var airspeedVelocity: Double {
+        return 160.0
+    }
+}
+```
+
+Extensions also allow us to extend define instance methods and type methods for types which you do not have access to the original source code. For example: 
```

Comment:
I'm not sure what `extend define instance methods` means. Did you mean extend **and** define?